msvc/scripts/genconsts.cs

```diff
@@ -0,0 +1,98 @@
+using System;
```

Comment:
1. It might be less code to read configure.ac?
2. More so, whatever lines we parse from it, probably should be in separate file or files.

Reply:
My thinking was that whatever winsetup produces should match whatever genconsts produces. Are you suggesting making winsetup produce a file to read from instead of parsing stdout? That seems reasonable

Reply:
I was thinking of extracting lines from configure.ac into, say, monoversion.txt.
Have one small file for everyone to read and easily parse.
But maybe not worth it. Maybe configure.ac is too strict and must have those lines.

Your idea is also good though, similar, but less.

I just kinda think that "files" tend to have more stable formats than "stdout".

For example, it should be possible to debug, by removing `echo off`, without breaking stuff.

But what you have is ok. I can debug just by running the thing directly.

Reply:
Also, your comment said you generate the file if it doesn't exist. That sounds wrong.  You generate it if it is out of date or incorrect. Such as, worst but common case, generate it always, compare to what you have,  and if different, move it over. Hopefully that isn't slow.

Reply:
Like this in winsetup.bat:

`:: If the file is different, replace it.
fc %monotemp% %config_h% >nul || move /y %monotemp% %config_h%`

but in a more elegant efficient  C# way. (The cmd way is inefficient in that it creates an extra process, for example.)

Reply:
I was also thinking -- what I really said -- is you could read configure.ac and duplicate winsetup.bat probably in far fewer lines. But that does lose synchronicity, granted. More code to achieve sharing logic is better than less code and duplicated logic. But you also fixup the number, so the logic doesn't match. Anyway, this is much ado about little.